File: 프로그래머스/0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.py
def solution(arr, k):
    answer = []

    for a in arr:
        if a not in answer:
            answer.append(a)
        if len(answer) == k:
            break
    
    if len(answer) < k:
        answer = answer + ([-1] * (k - len(answer)))
        
    return answer

File: 프로그래머스/0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int arr[], size_t arr_len, int k) {
    int flag[100001] = {0}; 
    int* answer = (int*)malloc(k * sizeof(int));
    int idx = 0;
    
    for(int i = 0; i < arr_len; i++){
        if(flag[arr[i]] == 1) continue;
        if(idx == k) break;
        
        answer[idx++] = arr[i];
        flag[arr[i]] = 1;
    }
    
    if(idx < k){
        for(; idx < k; idx++){
            answer[idx] = -1;
        }
    }
    
    return answer;
}